wire-ios-sync-engine/Source/Calling/WireCallCenterV3.swift

```diff
@@ -459,6 +464,48 @@ extension WireCallCenterV3 {
 
 }
 
+// MARK: - Call ending helpers
+
+extension WireCallCenterV3 {
+
+    /// This is a short term solution for 1:1 calls via SFT.
+    /// We treat 1:1 calls as conferences (via SFT) if `useSFTForOneToOneCalls` from the `conferenceCalling` feature is `true`.
+    /// If the other user hangs up, we should end the call for the self user.
+    /// More info (Option 1): https://wearezeta.atlassian.net/wiki/spaces/PAD/pages/1314750477/2024-07-29+1+1+calls+over+SFT
+    private func shouldEndCall(conversationId: AVSIdentifier, participants: [AVSCallMember]) -> Bool {
+        guard let context = uiMOC,
+              let conversation = ZMConversation.fetch(
+                with: conversationId.identifier,
+                domain: conversationId.domain,
+                in: context),
+              conversation.conversationType == .oneOnOne,
+              callSnapshots[conversationId]?.callState == .established
+        else {
+            return false
+        }
+
+        switch conversation.messageProtocol {
+        case .mls:
+            return shouldEndCallForMLS(participants: participants)
+        case .mixed, .proteus:
+            return shouldEndCallForProteus(participants: participants)
+        }
+    }
+
+    private func shouldEndCallForMLS(participants: [AVSCallMember]) -> Bool {
+        guard participants.count == 2,
+              participants[1].audioState == .connecting else {
```

Comment:
What is special about participants[1]?

Reply:
I guess we're assuming the 2nd participant is the other user? And if the other user audio state is `connecting` then the call should end? I think it may be better to filter out the self user in that case

Reply:
As I understood, the self user is always the first member in the payload from AVS. The 2nd participant is the other user

Reply:
Theoretically we can filter out the self-user, but this is a short term solution. Later there will be a proper solution. https://wearezeta.atlassian.net/wiki/spaces/PAD/pages/1314750477/2024-07-29+1+1+calls+over+SFT#Short-term-solution
To be able to filter out self user, we need to fetch it and compare with `AVSCallMember.client`. I can affect performance. @David-Henner  @johnxnguyen Do you insist on this?

Reply:
Good point about performance, I'd also prefer not to fetch the self user. Though I think it should be possible to avoid a big hit on performance by fetching the self user only once and caching it.
If you'd prefer to avoid it anyway because it's a short term solution, then go for it 👍 But perhaps we could add a comment to explain that we assume the first member to be the self user ?

Reply:
I'm fine with it as is for a temporary fix but a comment would be helpful.

Reply:
@David-Henner done